src/vulnpy/django/vulnerable_asgi.py

```diff
@@ -0,0 +1,93 @@
+from django.http import HttpResponse
+import django
+import urllib.parse
+
+try:
+    from django.conf.urls import url as compat_url
+except ImportError:
+    from django.urls import re_path as compat_url
+
+from vulnpy.common import get_template
+from vulnpy.trigger import TRIGGER_MAP, get_trigger
+
+
+async def _get_user_input(request):
+    # Django 3.* doesn't have a body for GET requests
```

Comment:
I don't understand this - HTTP GET requests in general do not have a request body. Do you get an error when you use `request.GET.get("user_input", "")` in django 3?

Reply:
In Django 3 it gets added to the header instead of the GET dictionary.

Reply:
I'm not able to reproduce this behavior even when I use django 3. The reason why I'm against it is because it's not the standard way to retrieve URL parameters in Django. What steps are you taking to get an error here?

Reply:
ah, python 3.7. I'll try that

Reply:
I messed up and forgot to actually commit the changes. Pipeline is rerunning now 👀

Reply:
They are all showing different things but it's all because request.GET is empty

Reply:
alright! I got it now. This is very strange; it's strangely similar to the problem with POST, because when I stand up the server live, I don't run into the issue (seems to only be a problem when running with pytest / AsyncClient)

I guess we should merge this. I really really think that the problems we're seeing are somehow related to test setup / configuration, not with the actual application or django itself. I'll give you the ✅, but I hope we can resolve some of these strange issues in the near future

Reply:
I actually see what you mean, I'm mostly suspecting the AsyncClient. I'll take a final look tomorrow

Reply:
My suspicions where correct it's the AsyncClient at least for that particular problem, I'm gonna add one more condition to the if statement so we can bypass this, I really don't see a better way it's not something configurable it's just how AsyncClient works in Django 3 😕.

Reply:
Thanks for mentioning it Peter I shouldn't have made such assumptions.